source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -513,11 +513,12 @@ ValidationResults ContextImpl::customVerifyCertChain(
     return {ValidationResults::ValidationStatus::Failed, SSL_AD_INTERNAL_ERROR, absl::nullopt};
   }
   ASSERT(cert_validator_);
+  const char* host_name = SSL_get_servername(ssl, TLSEXT_NAMETYPE_host_name);
   // Do not provide async callback here, but defer its creation to extended_socket_info if the
   // validation is async.
   ValidationResults result = cert_validator_->doVerifyCertChain(
       *cert_chain, nullptr, extended_socket_info, transport_socket_options, *SSL_get_SSL_CTX(ssl),
-      {}, SSL_is_server(ssl));
+      {}, SSL_is_server(ssl), (host_name == nullptr ? "" : host_name));
```

Comment:
`absl::NullSafeStringView()` will do this check and is used elsewhere to wrap boringssl values.

Reply:
done